packages/slate-react/src/utils/dom.ts

```diff
@@ -174,3 +174,13 @@ export const getPlainText = (domNode: DOMNode) => {
 
   return text
 }
+
+export const setReverseDomSelection = (
+  domRange: DOMRange,
+  domSelection: Selection
+) => {
+  const newRange = domRange.cloneRange()
+  newRange.collapse()
+  domSelection.addRange(newRange)
+  domSelection.extend(domRange.startContainer, domRange.startOffset)
```

Comment:
Hmm it would be better if this was all done in one place. Ideally `toDOMRange` could do this itself, rather than needing to patch what it returns, although that would require having it modify the global selection 🤔

I can't recall whether storing a backwards selection range and then re-adding it via `addRange` works, though, which would make that impossible.

Reply:
I found no way to create a reverse range according to https://dom.spec.whatwg.org/#concept-range-bp-set.

use `Selection.extend` is only way I found to make it work

Reply:
but if you make a backwards selection through `extend`, and then store the range to later set via `domSelection.addRange()`, would that work?

It's a bit of a round-about way to do it, but I'm suggesting we avoid splitting the set selection logic (any new code expecting `toDOMRange` to return a valid selection might also trigger a bug with backwards selections).

Reply:
https://w3c.github.io/selection-api/#definition

there is only one selection in document as above document. make a backwards selection through `Selection.extend` will change the selection of the document and will trigger onSelection change event.

I understand what your concern but it is dom API limit us.

Reply:
This seems like a useful concern to capture here for posterity. Could you add a note here to why we need to clone and mutate the range in this way?

Reply:
@TheSpyder  does leaving it as a comment work for you? We might be able to do something to improve it at a later time if it 💥  but this seems to work well

Reply:
I realise there's only one selection in a document, I was talking about storing the current selection, mutating it to the range desired, then storing the range and restoring the original selection. It's probably a bad idea but would keep everything contained.

I'd like to see a similar comment on `toDOMRange`, as that's where people are likely to look first for selection-related things rather than the `dom` util module.

Reply:
thank for preview and i have add comments for `setReverseDomSelection` and `toDOMRange`